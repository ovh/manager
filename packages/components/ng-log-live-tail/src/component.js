import template from './template.html';
import controller from './controller';

export default {
  controller,
  template,
  transclude: {
    tile: '?tile',
    description: '?description',
  },
  bindings: {
    logApiUrl: '<',
    logKind: '<',
    logKeys: '<',
  },
};
