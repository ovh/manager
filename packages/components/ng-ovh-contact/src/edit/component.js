import controller from './controller';
import template from './template.html';

export default {
  require: {
    ovhContactCtrl: '^ovhContact',
  },
  template,
  controller,
  bindings: {
    options: '=?ovhContactChoiceOptions',
    customList: '=?ovhContactChoiceCustomList',
  },
};
