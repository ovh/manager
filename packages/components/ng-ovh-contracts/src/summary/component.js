import template from './template.html';

const component = {
  bindings: {
    contracts: '<items',
    inlineContracts: '<inlineItems',
    model: '=',
    name: '@?',
  },
  template,
};

export default component;
