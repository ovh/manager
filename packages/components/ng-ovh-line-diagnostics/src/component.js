import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    lineNumber: '@',
    serviceName: '@',
  },
  template,
  controller,
};
