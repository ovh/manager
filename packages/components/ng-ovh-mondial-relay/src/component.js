import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    ngModel: '=?',
    userService: '=?',
    mondialRelayService: '=?',
  },
  template,
  controller,
};
