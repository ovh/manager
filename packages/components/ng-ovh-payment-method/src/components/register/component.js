import controller from './controller';
import template from './index.html';

export default {
  name: 'ovhPaymentMethodRegister',
  controller,
  template,
  bindings: {
    defaultPaymentMethodType: '@?',
    defaultPaymentMethodIntegration: '@?',
    model: '<',
    modelName: '@?',
    automaticSelect: '<?',
    automaticDefault: '<?',
    paymentMethodTypesOrder: '<?',
    paymentMethodTypesPerLine: '<?',
    registeredPaymentMethods: '<?',
    unregisteredPaymentMethods: '<?',
    excludedPaymentMethods: '<?',
    showSetAsDefaultChoice: '<?',
    forceSetAsDefaultChoice: '<?',
    onInitializationError: '&?',
    onInitialized: '&?',
    showDefaultExplanationTexts: '<?',
    defaultExplanationTextsOptions: '<?',
    required: '<?',
  },
  transclude: {
    introductionText: '?ovhPaymentMethodRegisterIntroductionText',
    explanationText: '?ovhPaymentMethodRegisterExplanationText',
    creditCardExplanationText:
      '?ovhPaymentMethodRegisterCreditCardExplanationText',
    sepaExplanationText: '?ovhPaymentMethodRegisterSepaExplanationText',
    paypalExplanationText: '?ovhPaymentMethodRegisterPaypalExplanationText',
    rupayExplanationText: '?ovhPaymentMethodRegisterRupayExplanationText',
  },
};
