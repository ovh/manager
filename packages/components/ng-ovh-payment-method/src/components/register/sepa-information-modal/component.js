import template from './template.html';

export default {
  template,
  bindings: {
    onConfirm: '&',
    isModalOpened: '<',
  },
};
