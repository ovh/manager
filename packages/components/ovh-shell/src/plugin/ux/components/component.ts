export interface IUxComponentOptions {
  visible?: boolean;
}

export interface IUxComponent {
  toggleVisibility: () => void;
  show: () => void;
  hide: () => void;
  getVisibility: () => boolean;
  onVisibilityChange: (callback: CallableFunction) => void;
}

export class UXComponent implements IUxComponent {
  private visible: boolean;

  protected listeners: CallableFunction[] = [];

  constructor(options: IUxComponentOptions = {}) {
    this.visible = options.visible || false;
  }

  public reset(options: IUxComponentOptions = {}) {
    this.setVisibility(options.visible || false);
  }

  public setVisibility = (visibility: boolean) => {
    this.visible = visibility;
    this.triggerListeners();
  };

  public toggleVisibility(): void {
    this.setVisibility(!this.visible);
  }

  public show(): void {
    this.setVisibility(true);
  }

  public hide(): void {
    this.setVisibility(false);
  }

  public getVisibility(): boolean {
    return this.visible;
  }

  public onVisibilityChange(callback: CallableFunction): void {
    this.listeners.push(callback);
  }

  private triggerListeners() {
    this.listeners.forEach((listener) => {
      listener();
    });
  }
}
