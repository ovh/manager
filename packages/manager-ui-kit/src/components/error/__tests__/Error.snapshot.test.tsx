import { vitest } from 'vitest';
import { Error } from '../Error.component';
import { ErrorObject } from '../Error.props';
import {
  mockGetEnvironment,
  renderWithContext,
} from '../../../utils/Test.utils';

mockGetEnvironment.mockResolvedValue({
  applicationName: 'test-application',
});

describe('ErrorBanner Snapshot Tests', () => {
  beforeEach(() => {
    vitest.clearAllMocks();
  });

  it('should match snapshot with default 404 error', () => {
    const defaultError: ErrorObject = {
      status: 404,
      data: { message: 'Service not found' },
      headers: { 'x-ovh-queryid': '123456789' },
    };
    const { container } = renderWithContext({
      children: <Error error={defaultError} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with 401 unauthorized error', () => {
    const unauthorizedError: ErrorObject = {
      status: 401,
      data: { message: 'Unauthorized access' },
      headers: { 'x-ovh-queryid': '987654321' },
    };
    const { container } = renderWithContext({
      children: <Error error={unauthorizedError} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with 500 server error', () => {
    const serverError: ErrorObject = {
      status: 500,
      data: { message: 'Internal server error' },
      headers: { 'x-ovh-queryid': '555666777' },
    };
    const { container } = renderWithContext({
      children: <Error error={serverError} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with error without query ID', () => {
    const errorWithoutQueryId: ErrorObject = {
      status: 404,
      data: { message: 'Page not found' },
      headers: {},
    };
    const { container } = renderWithContext({
      children: <Error error={errorWithoutQueryId} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with error without data message', () => {
    const errorWithoutMessage: ErrorObject = {
      status: 500,
      data: {},
      headers: { 'x-ovh-queryid': '111222333' },
    };
    const { container } = renderWithContext({
      children: <Error error={errorWithoutMessage} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with custom label tracking', () => {
    const error: ErrorObject = {
      status: 404,
      data: { message: 'Custom error message' },
      headers: { 'x-ovh-queryid': '444555666' },
    };
    const { container } = renderWithContext({
      children: <Error error={error} labelTracking="custom-tracking-label" />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with custom className', () => {
    const error: ErrorObject = {
      status: 404,
      data: { message: 'Test error' },
      headers: { 'x-ovh-queryid': '777888999' },
    };
    const { container } = renderWithContext({
      children: <Error error={error} className="custom-error-class" />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with all props', () => {
    const error: ErrorObject = {
      status: 403,
      data: { message: 'Forbidden access' },
      headers: { 'x-ovh-queryid': '000111222' },
    };

    const onRedirectHome = vitest.fn();
    const onReloadPage = vitest.fn();
    const { container } = renderWithContext({
      children: (
        <Error
          error={error}
          onRedirectHome={onRedirectHome}
          onReloadPage={onReloadPage}
          labelTracking="full-props-test"
          className="full-test-class"
        />
      ),
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with minimal error object', () => {
    const minimalError: ErrorObject = {
      status: 404,
      data: {},
      headers: {},
    };
    const { container } = renderWithContext({
      children: <Error error={minimalError} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with complex error message', () => {
    const complexError: ErrorObject = {
      status: 500,
      data: {
        message:
          'This is a very long error message that might contain special characters like é, ñ, or symbols like @#$%^&*() and should be displayed properly in the error banner component',
      },
      headers: { 'x-ovh-queryid': 'complex123' },
    };
    const { container } = renderWithContext({
      children: <Error error={complexError} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with null error', () => {
    const { container } = renderWithContext({
      children: <Error error={null} />,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
