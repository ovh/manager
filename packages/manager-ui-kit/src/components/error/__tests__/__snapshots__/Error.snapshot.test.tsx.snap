// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBanner Snapshot Tests > should match snapshot with 401 unauthorized error 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
    Oups ! Une erreur interne est survenue
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Une erreur s'est produite, vous pouvez nous contacter avec le code d'erreur ci dessous.
  </h2>
  <div
    class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9 w-full"
    data-ods="message"
  >
    <span
      class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
      data-ods="message-icon"
      role="presentation"
    />
    <div
      class="_message-body_fe7x3_2"
      data-ods="message-body"
    >
      <p
        class="m-0 p-O"
      >
        <span
          class="block"
        >
          <strong>
            Unauthorized access
          </strong>
        </span>
        <span
          class="block py-[10px]"
        >
          Code d'erreur : 
          987654321
        </span>
      </p>
    </div>
  </div>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with 500 server error 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
    Oups ! Une erreur interne est survenue
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Une erreur s'est produite, vous pouvez nous contacter avec le code d'erreur ci dessous.
  </h2>
  <div
    class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9 w-full"
    data-ods="message"
  >
    <span
      class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
      data-ods="message-icon"
      role="presentation"
    />
    <div
      class="_message-body_fe7x3_2"
      data-ods="message-body"
    >
      <p
        class="m-0 p-O"
      >
        <span
          class="block"
        >
          <strong>
            Internal server error
          </strong>
        </span>
        <span
          class="block py-[10px]"
        >
          Code d'erreur : 
          555666777
        </span>
      </p>
    </div>
  </div>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with all props 1`] = `
<section
  class="full-test-class"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
    Oups ! Une erreur interne est survenue
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Une erreur s'est produite, vous pouvez nous contacter avec le code d'erreur ci dessous.
  </h2>
  <div
    class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9 w-full"
    data-ods="message"
    data-tracking="full-props-test"
  >
    <span
      class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
      data-ods="message-icon"
      role="presentation"
    />
    <div
      class="_message-body_fe7x3_2"
      data-ods="message-body"
    >
      <p
        class="m-0 p-O"
      >
        <span
          class="block"
        >
          <strong>
            Forbidden access
          </strong>
        </span>
        <span
          class="block py-[10px]"
        >
          Code d'erreur : 
          000111222
        </span>
      </p>
    </div>
  </div>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with complex error message 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
    Oups ! Une erreur interne est survenue
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Une erreur s'est produite, vous pouvez nous contacter avec le code d'erreur ci dessous.
  </h2>
  <div
    class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9 w-full"
    data-ods="message"
  >
    <span
      class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
      data-ods="message-icon"
      role="presentation"
    />
    <div
      class="_message-body_fe7x3_2"
      data-ods="message-body"
    >
      <p
        class="m-0 p-O"
      >
        <span
          class="block"
        >
          <strong>
            This is a very long error message that might contain special characters like é, ñ, or symbols like @#$%^&*() and should be displayed properly in the error banner component
          </strong>
        </span>
        <span
          class="block py-[10px]"
        >
          Code d'erreur : 
          complex123
        </span>
      </p>
    </div>
  </div>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with custom className 1`] = `
<section
  class="custom-error-class"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Page introuvable.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Erreur 404. La page que vous recherchez semble introuvable.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with custom label tracking 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Page introuvable.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Erreur 404. La page que vous recherchez semble introuvable.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with default 404 error 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Page introuvable.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Erreur 404. La page que vous recherchez semble introuvable.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with error without data message 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
    Oups ! Une erreur interne est survenue
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Une erreur s'est produite, vous pouvez nous contacter avec le code d'erreur ci dessous.
  </h2>
  <div
    class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9 w-full"
    data-ods="message"
  >
    <span
      class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
      data-ods="message-icon"
      role="presentation"
    />
    <div
      class="_message-body_fe7x3_2"
      data-ods="message-body"
    >
      <p
        class="m-0 p-O"
      >
        <span
          class="block py-[10px]"
        >
          Code d'erreur : 
          111222333
        </span>
      </p>
    </div>
  </div>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with error without query ID 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Page introuvable.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Erreur 404. La page que vous recherchez semble introuvable.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with minimal error object 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Page introuvable.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
    Erreur 404. La page que vous recherchez semble introuvable.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;

exports[`ErrorBanner Snapshot Tests > should match snapshot with null error 1`] = `
<section
  class="mx-auto w-full h-full max-w-[600px] overflow-hidden mx-autogrid p-5"
>
  <img
    alt="OOPS"
    class="w-full"
    src="/public/assets/error-banner-oops.png"
  />
  <h1
    class="_text--heading-1_1aw48_13 pt-2"
    data-ods="text"
  >
     
    Oups ! Une erreur est survenue.
  </h1>
  <h2
    class="_text--heading-2_1aw48_19 pt-1 pb-2"
    data-ods="text"
  >
     
    Une erreur s'est produite, veuillez réessayer plus tard.
  </h2>
  <div
    class="overflow-hidden mt-5 py-2"
  >
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-home"
      type="button"
    >
      Retour à la page d'accueil
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 error-template-actions w-full"
      data-ods="button"
      data-testid="error-template-action-reload"
      type="button"
    >
      Réessayer
    </button>
  </div>
</section>
`;
